datafusion/core/tests/sql/decimal.rs

```diff
@@ -1,915 +0,0 @@
-        "| 0.000030 | 0.000000000003 | 4   | true  | 0.0000320 |",
-        "| 0.000030 | 0.000000000003 | 5   | false | 0.0000350 |",
-        "| 0.000030 | 0.000000000003 | 5   | true  | 0.0000110 |",
-        "| 0.000040 | 0.000000000004 | 5   | true  | 0.0000440 |",
-        "| 0.000040 | 0.000000000004 | 12  | false | 0.0000400 |",
-        "| 0.000040 | 0.000000000004 | 14  | true  | 0.0000400 |",
-        "| 0.000040 | 0.000000000004 | 8   | false | 0.0000440 |",
-        "| 0.000050 | 0.000000000005 | 9   | true  | 0.0000520 |",
-        "| 0.000050 | 0.000000000005 | 4   | true  | 0.0000780 |",
-        "| 0.000050 | 0.000000000005 | 8   | false | 0.0000330 |",
-        "| 0.000050 | 0.000000000005 | 100 | true  | 0.0000680 |",
-        "| 0.000050 | 0.000000000005 | 1   | false | 0.0001000 |",
-        "+----------+----------------+-----+-------+-----------+",
-    ];
-    assert_batches_eq!(expected, &actual);
-
-    // logic operation: gteq
-    let sql = "select * from decimal_simple where c1 >= 0.00002";
-    let actual = execute_to_batches(&ctx, sql).await;
-    assert_eq!(
-        &DataType::Decimal128(10, 6),
-        actual[0].schema().field(0).data_type()
-    );
-    let expected = vec![
-        "+----------+----------------+-----+-------+-----------+",
-        "| c1       | c2             | c3  | c4    | c5        |",
-        "+----------+----------------+-----+-------+-----------+",
-        "| 0.000020 | 0.000000000002 | 2   | true  | 0.0000250 |",
-        "| 0.000020 | 0.000000000002 | 3   | false | 0.0000190 |",
-        "| 0.000030 | 0.000000000003 | 4   | true  | 0.0000320 |",
-        "| 0.000030 | 0.000000000003 | 5   | false | 0.0000350 |",
-        "| 0.000030 | 0.000000000003 | 5   | true  | 0.0000110 |",
-        "| 0.000040 | 0.000000000004 | 5   | true  | 0.0000440 |",
-        "| 0.000040 | 0.000000000004 | 12  | false | 0.0000400 |",
-        "| 0.000040 | 0.000000000004 | 14  | true  | 0.0000400 |",
-        "| 0.000040 | 0.000000000004 | 8   | false | 0.0000440 |",
-        "| 0.000050 | 0.000000000005 | 9   | true  | 0.0000520 |",
-        "| 0.000050 | 0.000000000005 | 4   | true  | 0.0000780 |",
-        "| 0.000050 | 0.000000000005 | 8   | false | 0.0000330 |",
-        "| 0.000050 | 0.000000000005 | 100 | true  | 0.0000680 |",
-        "| 0.000050 | 0.000000000005 | 1   | false | 0.0001000 |",
-        "+----------+----------------+-----+-------+-----------+",
-    ];
-    assert_batches_eq!(expected, &actual);
-
-    Ok(())
-}
-
-#[tokio::test]
-async fn decimal_arithmetic_op() -> Result<()> {
-    let ctx = SessionContext::new();
-    register_decimal_csv_table_by_sql(&ctx).await;
-    // add
-    let sql = "select c1+1 from decimal_simple"; // add scalar
-    let actual = execute_to_batches(&ctx, sql).await;
-    // array decimal(10,6) +  scalar decimal(20,0) => decimal(21,6)
-    assert_eq!(
-        &DataType::Decimal128(27, 6),
```

Comment:
I noticed that the comment expects a precision `21`, but the assertion is `27`. I am not sure if it's a typo in the comment, or if something is wrong with the type. I moved this test to `decimal.slt` as is.